pkg/controller/orchestrator/orchestrator_reconcile.go

```diff
@@ -0,0 +1,470 @@
+				log.V(2).Info("Failed to register %s with orchestrator", "host", host, "error", err)
+				errMsg += fmt.Sprintf(" failed to register %s", host)
+			}
+		} else {
+			instances = append(instances, *inst)
+		}
+	}
+
+	// remove all instances from orchestrator that does not exists in k8s
+	for _, inst := range instSet {
+		if i := instances.GetInstance(inst.Key.Hostname); i == nil {
+			if err = ou.orcClient.Forget(inst.Key.Hostname, inst.Key.Port); err != nil {
+				log.V(2).Info("Failed to forget %s with orchestrator", "host", i.Key.Hostname, "error", err)
+				errMsg += fmt.Sprintf(" failed to forget %s", inst.Key.Hostname)
+			}
+		}
+	}
+
+	if len(errMsg) != 0 {
+		err = errors.New(errMsg)
+	}
+	return instances, err
+}
+
+func (ou *orcUpdater) getRecoveriesToAck(recoveries []orc.TopologyRecovery) []orc.TopologyRecovery {
+	toAck := []orc.TopologyRecovery{}
+
+	if len(recoveries) == 0 {
+		return toAck
+	}
+
+	i, find := condIndexCluster(ou.cluster, api.ClusterConditionReady)
+	if !find || ou.cluster.Status.Conditions[i].Status != core.ConditionTrue {
+		log.Info("Cluster is not ready for ack.")
+		return toAck
+	}
+
+	if time.Since(ou.cluster.Status.Conditions[i].LastTransitionTime.Time).Minutes() < healtyMoreThanMinutes {
+		log.Info(fmt.Sprintf("Stateful set is not ready more then %d minutes. Don't ack.", healtyMoreThanMinutes))
+		return toAck
+	}
+
+	for _, recovery := range recoveries {
+		if !recovery.Acknowledged {
+			// skip if it's a new recovery, recovery should be older then <healtyMoreThanMinutes> minutes
+			startTime, err := time.Parse(time.RFC3339, recovery.RecoveryStartTimestamp)
+			if err != nil {
+				log.Error(err, "Can't parse time: %s for audit recovery.",
+					"recovery", recovery,
+				)
+				continue
+			}
+			if time.Since(startTime).Minutes() < healtyMoreThanMinutes {
+				// skip this recovery
+				log.Error(nil, "Tries to recover to soon.", "recovery", recovery)
+				continue
+			}
+
+			toAck = append(toAck, recovery)
+		}
+	}
+	return toAck
+}
+
+func (ou *orcUpdater) acknowledgeRecoveries(toAck []orc.TopologyRecovery) error {
+	comment := fmt.Sprintf("Statefulset '%s' is healty for more then %d minutes",
+		ou.cluster.GetNameForResource(api.StatefulSet), healtyMoreThanMinutes,
+	)
+
+	// acknowledge recoveries
+	for _, recovery := range toAck {
+		if err := ou.orcClient.AckRecovery(recovery.Id, comment); err != nil {
+			return err
+		}
+		ou.recorder.Event(ou.cluster, eventNormal, "RecoveryAcked",
+			fmt.Sprintf("Recovery with id %d was acked.", recovery.Id))
+	}
+
+	return nil
+}
+
+func (ou *orcUpdater) updateStatusForRecoveries(recoveries []orc.TopologyRecovery) {
+	var unack []orc.TopologyRecovery
+	for _, recovery := range recoveries {
+		if !recovery.Acknowledged {
+			unack = append(unack, recovery)
+		}
+	}
+
+	if len(unack) > 0 {
+		msg := getRecoveryTextMsg(unack)
+		ou.cluster.UpdateStatusCondition(api.ClusterConditionFailoverAck,
+			core.ConditionTrue, "pendingFailoverAckExists", msg)
```

Comment:
`pendingFailoverAckExists` and `noPendingFailoverAckExists` should be in UpperCamelCaseNotation